### PR TITLE
Migrate to Go modules and upgrade to Go 1.19

### PR DESCRIPTION
**Pull request by andrewsheng2**
Some dependencies have been updated so that the minimum version of Go is now 1.19. This repo currently doesn't use Go modules and our Terraform Docker image doesn't properly pull the right dependencies, so building the image currently does not work. Migrate dependency management over to Go modules to make the update process easier.

J=SB-10399
TEST=manual

Ran `GOOS=linux GOARCH=amd64 go build` and made a temporary Terraform image using the compiled binary and checked that it applying pipelines to a local TeamCity still works

### CODE REVIEW

**Review by yzhangyext — APPROVED** [review]
Nice! Can be a followup, but it would be nice to also test the local provider workflow and update the instructions:
https://gerrit.yext.com/plugins/gitiles/alpha/+/master/ci/images/terraform/README.md

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+go 1.19
+
+require (
+	github.com/hashicorp/terraform v0.12.2
+	github.com/robfig/cron v1.2.0
+	github.com/yext/go-teamcity v0.5.3-0.20230323164551-08f28b0ad89b
+)
+
+require (
```
Comment (bsaavedra17) [inline go.mod:11]: Should both of these require statements be merged? Or was there a particular reason they were made separate?

**Review by bsaavedra17 — COMMENTED** [review]
Left a small comment regarding the require statements, otherwise looks good!

File: go.mod (hunk 1 above)
Comment (andrewsheng2) [reply go.mod:11]: It looks like that is the expected format for go.mod. The first section lists all direct dependencies used in the code, and the second section has all indirect dependencies. This is also the case for the go.mod file in alpha

Comment (bsaavedra17) [reply go.mod:11]: Looks good to me then, thanks for the explanation!

**Review by bsaavedra17 — APPROVED** [review]

**Pull request merged**